src/math/FixedKernel.cc

```diff
@@ -59,6 +59,45 @@ std::shared_ptr<Kernel> FixedKernel::clone() const {
     return retPtr;
 }
 
+std::shared_ptr<Kernel> FixedKernel::resized(int width, int height) const {
+    if ((width <= 0) || (height <= 0)) {
+        std::ostringstream os;
+        os << "Cannot create FixedKernel with dimensions (" << width << ", " << height << "). ";
+         throw LSST_EXCEPT(pexExcept::InvalidParameterError, os.str());
+    }
+    if ((width - this->getWidth()) % 2 || (height - this->getHeight()) % 2) {
+        std::ostringstream os;
+        os << "Cannot resize FixedKernel from ("
+        <<  this->getWidth() << ", " << this->getHeight() << ") to (" << width << ", " << height
+        << "), because at least one dimension would change by an odd value.";
+        throw LSST_EXCEPT(pexExcept::InvalidParameterError, os.str());
+    }
+
+    lsst::afw::geom::Box2I bboxNew(lsst::afw::geom::Point2I((1 - width)/2, (1 - height)/2),
+                                   lsst::afw::geom::Extent2I(width, height));
+    std::shared_ptr<image::Image<Pixel>> imNew(new image::Image<Pixel>(bboxNew));
+
+    lsst::afw::geom::Box2I bboxIntersect = this->getBBox();
+    bboxIntersect.clip(bboxNew);
+
+    // Kernel (this) and member image (this->_image) do not always have same XY0.
+    // Member image of resized kernel will not have same BBox as orig,
+    // but BBox of member image is ignored by the kernel.
+    int offsetX = this->_image.getX0() - this->getBBox().getMinX();
+    int offsetY = this->_image.getY0() - this->getBBox().getMinY();
+    lsst::afw::geom::Box2I bboxIntersectShifted = lsst::afw::geom::Box2I(
+        lsst::afw::geom::Point2I(bboxIntersect.getMinX() + offsetX,
+                                 bboxIntersect.getMinY() + offsetY),
+        bboxIntersect.getDimensions());
+    image::Image<Pixel> imIntersect = image::Image<Pixel>(_image, bboxIntersectShifted);
+
+    imNew->assign(imIntersect, bboxIntersect);
+
+    std::shared_ptr<image::Image<Pixel> const> imConst = imNew;
+    std::shared_ptr<Kernel> retPtr(new FixedKernel(*imConst));
```

Comment:
Thank you.  I needed imNew because I'm only assigning  pixels on the new image where the old image and new image overlap.  I don't think we have `Image` constructor that can do that.  I did remove the imConst though (make_shared fixed the casting issue I was having before).